core/src/main/scala/org/bykn/bosatsu/rankn/Infer.scala

```diff
@@ -1462,6 +1479,130 @@ object Infer {
       }
     }
 
+    // noshadow must include any free vars of args
+    def liftQuantification[A](
```

Comment:
Should add direct tests of this in simple cases and possibly property checks. For instance, if there is a quantification on any arg, it should be lifted up. Or rather the result should not have quantification on any argument.